mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/dependencies/DependencyResolver.java

```diff
@@ -30,21 +32,41 @@ private void addStrategies(Model model) {
         final StrategyDependencyResolver strategyDependencyResolver = new StrategyDependencyResolver();
         model.addDependency(smartTestingProviderDependency());
         final Map<String, Dependency> dependencies = strategyDependencyResolver.resolveDependencies();
-        final Map<String, String> strategyMismatch = new HashMap<>();
-        for (final String definedStrategy : strategies) {
+        final List<String> strategyMismatch = new ArrayList<>();
+        final List<String> registeredStrategies = new ArrayList<>();
+        for (int i=0; i < strategies.length; i++) {
+            String definedStrategy = strategies[i];
             if (!dependencies.containsKey(definedStrategy)) {
                 final String closestMatch = stringSimilarityCalculator.findClosestMatch(definedStrategy, dependencies.keySet());
-                strategyMismatch.put(definedStrategy, closestMatch);
+                if (configuration.isAutocorrect()) {
```

Comment:
The whole method `addStrategies` is pretty long now and the if hierarchy is quite complicated. Do you think that it could be sliced to multiple methods?

Reply:
I tried but then you get methods with 4 or 5 parameters, so I decided that it was better to maintain together.